lib/services/elasticsearch.js

```diff
@@ -561,6 +564,41 @@ function ElasticSearch(kuzzle, options, config) {
       .then(ids => {
         return new Promise((resolve, reject) => {
           async.each(ids, (id, callback) => {
+            bodyBulk.push({update: {_index: data.index, _type: data.type, _id: id}});
+            bodyBulk.push({doc: {_kuzzle_info: { active: false, deletedAt: Date.now() }}});
+            callback();
+          }, () => {
+            if (bodyBulk.length === 0) {
+              return resolve({ids: []});
+            }
+
+            this.client.bulk({body: bodyBulk})
+              .then(() => refreshIndexIfNeeded.call(this, data, {ids: ids}))
+              .then(result => resolve(result))
+              .catch(error => reject(this.formatESError(error)));
+          });
+        });
+      });
+  };
+
+  /**
+   * Delete all document that match the given filter from the trash
+   * @param requestObject
+   */
+  this.deleteByQueryFromTrash = function elasticsearchDeleteByQueryFromTrash (requestObject) {
```

Comment:
:thinking: So, you mean that the GarbageCollector is responsible of ensuring that only the documents from the trash are deleted, which is :ok: to me. But, in this case, I think this function should be named `deleteByQuery` rather that `deleteByQueryFromTrash` (since the current name suggests that this particular function is in charge of deleting documents from the trash, which is not the case).